### ISSUE TITLE
No LCD-text-antialiasing anymore since Electron 1.2.0 (since the Chrome 51 update)

### ISSUE THREAD

**Issue by krpano** [electron/electron#6344]
Since Electron 1.2.0, that means since the update from Chrome 50 to Chrome 51, there no LCD-text-antialiasing anymore (tested only on Windows).

Here an animated GIF showing the difference (300% zoomed screenshots) - look at the 'To run...' text:
![lcd-text-antialiasing](https://cloud.githubusercontent.com/assets/5989766/16565403/46503f2a-420d-11e6-9573-da40f6abd453.gif)
The text is now less sharp and slightly blurred...

NWJS has the same problem since updating to Chrome 51, but Chrome 51 itself is fine.

Due my own research it seems that the problem might be somehow related to some page/window background-color and/or transparency settings... (Chrome renders automatically with grayscale-antialiasing on transparent elements).

**Comment by zcbenz** [comment]
I tested with Chrome 51, but it also seems to use greyscale antialiasing:
<img width="496" alt="screen shot 2016-07-05 at 10 03 14 am" src="https://cloud.githubusercontent.com/assets/639601/16571472/c8fb2040-4297-11e6-874f-ca8be526c8bd.png">

I verified that LCD text antialiasing is enabled.

**Comment by paulcbetts** [comment]
What happens if you set the background of your page to `#fff` explicitly

**Comment by krpano** [comment]
No, that doesn't help anymore - the 'page background-color workaround' was only working before Electron 1.2.0, but since this version there is always the grayscale-antialiasing...

I'm currently browsing the Chromium source for code related to this issue and found this here:
https://cs.chromium.org/chromium/src/ui/gfx/render_text.h?rcl=1467731912&l=810
Might this be related and is there a way to check if that 'subpixel_rendering_suppressed_' variable is set or not?

**Comment by paulcbetts** [comment]
@krpano Gotta use a debugger

**Comment by zcbenz** [comment]
@krpano I saw you said Chrome 51 itself worked fine, but I can also reproduce with Chrome 51, can you try again and verify you are using Chrome 51? If this is a bug of Chrome 51 we need to report it to upstream.

**Comment by krpano** [comment]
I have just tested again and yes - all Chrome versions are fine for me, I have tested Chrome 48 to Chrome 53-dev (all on Windows 7x64).

Here two screenshots:

Chrome 50:
![chrome50](https://cloud.githubusercontent.com/assets/5989766/16609398/3ce884f4-4354-11e6-9b97-88823b77b49c.png)
Chrome 51:
![chrome51](https://cloud.githubusercontent.com/assets/5989766/16609402/41986e92-4354-11e6-8301-99a80bf37669.png)

**Comment by krpano** [comment]
The strange thing is that this happens also to NWJS since they updated to Chrome 51 (and above).
So it seem that's somehow related to the way how Electron and NWJS are 'using' Chromium...

**Comment by zcbenz** [comment]
Can you try passing `--enable-lcd-text` and `--disable-gpu` when running Electron and see if it makes a difference?

**Comment by bpasero** [comment]
I can reproduce this with VS Code updating to Electron 1.2.6 on Windows 10.

**Comment by krpano** [comment]
Just tried --enable-lcd-text and --disable-gpu again - no difference.
I think I have already tried all kind of Chromium switches (passed directly and via appendSwitch), no difference at all...

**Comment by bpasero** [comment]
This turns out to be a blocker for us to go with Electron 1.2.x (and likely for anyone that cares about a non-blurry UI). Since I cannot reproduce this in standalone Chrome I suspect this is an issue only for Electron/nw (as pointed out already). 

Should we report this to Chrome to get some ideas where to look for?

**Comment by zcbenz** [comment]
@bpasero Yeah ideas from Chrome developers would be very helpful.

**Comment by bpasero** [comment]
I hope I was accurate in the issue: https://bugs.chromium.org/p/chromium/issues/detail?id=625961

**Comment by joaomoreno** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=612013

It points towards a device specific issue.

**Comment by krpano** [comment]
@joaomoreno There are several cases where Chrome could have grayscale or blurred font rendering - e.g. transparent page background, css 3d transformed elements, sub-pixel-positioned elements or fractional page scaling - but this issue here seems to be an independent problem and not be to related to that.

The issue in your [link](https://bugs.chromium.org/p/chromium/issues/detail?id=612013) with the very blurred fonts looks like an fractional page scaling + sub pixel-positioning problem (and depending on the exact case it's possible to work-around that problem by doing a manual pixel-grid-fitting of the related element positions and sizes).

**Comment by be5invis** [comment]
@krpano What's your screen DPI?
AFAIK Chrome turns off LCD AA when screen DPI is over 100%.

**Comment by gdamjan** [comment]
I get the same issue on Linux with the yakyak app (hangouts wrapper). 
I'm on ArchLinux, Intel HD Graphics 5500 - Broadwell, modesetting driver xorg-server 1.18.3, kernel 4.6, mesa 12.0.1.

Even the inspector (yakyak has an option to run it) doesn't have fonts rgb-antialised. The menu in yakyak on the other hand is lcd-antialised. http://imgur.com/g9Dt8gD

It's interesting that Chromium 51.0.2704.106 works fine.

**Comment by krpano** [comment]
In the new NW.js 0.16.0 release the problem is fixed now with Chromium 52.0.2743.82!

Maybe (hopefully) the problem will get fixed for Electron too when updating to that Chromium version.

**Comment by krpano** [comment]
The problem is still not fixed in Electron 1.3.0 - even with the Chromium 52.0.2743.82 update!

That means it must be also something else in Electron itself and not just the Chromium version...

**Comment by zcbenz** [comment]
@krpano Can you try whether passing `--disable-prefer-compositing-to-lcd-text` works?

**Comment by krpano** [comment]
@zcbenz - no sorry, doesn't work...

**Comment by joshaber** [comment]
FWIW the Chrome team responded to @bpasero's issue:

>Please file a bug with Electron. We do not support third party products built on top of chrome.

**Comment by krpano** [comment]
Great Thanks!
Working well now.

**Comment by gdamjan** [comment]
What does "system settings" mean here? I'm on ArchLinux, running KDE/Plasma, fontconfig has 10-sub-pixel-rgb.conf enabled, all applications are rgb subpixel antialised except for yakyak.

```
process.versions
{
   "ares" : "1.10.1-DEV",
   "electron" : "1.3.1",
   "v8" : "5.2.361.43",
   "uv" : "1.9.1",
   "atom-shell" : "1.3.1",
   "http_parser" : "2.7.0",
   "openssl" : "1.0.2h",
   "node" : "6.3.0",
   "modules" : "49",
   "zlib" : "1.2.8",
   "chrome" : "52.0.2743.82"
}
```

at the same time, atom 1.8.0 is fine

**Comment by binaryfunt** [comment]
This is still a problem for me. Windows 10, Electron v2.0.9, Chromium 61, 1920x1080 LCD monitor with 100% DPI.

Top image is Chrome 68, bottom is Electron:

![untitled-6](https://user-images.githubusercontent.com/14579418/45376907-f448f280-b5f0-11e8-9d06-2347f4645251.png)

Strangely, I've noticed GitHub Desktop (built in Electron, using Chromium 61 when I checked) _does_ have sub-pixel anti-aliasing. Is there is some way of enabling it?

**Comment by Slapbox** [comment]
Holy crap! Can we get @binaryfunt's tip into the docs somewhere? If it's there already, I never found it when I researched this in the past for quite a few hours.

**Comment by nornagon** [comment]
See #10955

**Comment by heronils** [comment]
It is now in the [FAQ](https://github.com/electron/electron/blob/master/docs/faq.md#the-font-looks-blurry-what-is-this-and-what-can-i-do)

**Comment by uotw** [comment]
> It is now in the [FAQ](https://github.com/electron/electron/blob/master/docs/faq.md#the-font-looks-blurry-what-is-this-and-what-can-i-do)
@heronils I'm still seeing the blurriness with electron@7.1.11. Using this code as recommended:
`mainWindow = new BrowserWindow({ backgroundColor: '#fff', ...` Interestingly it works fine using `npm start`, but packaged using electron-packager - the blurriness returns:

<img width="760" alt="npmStart" src="https://user-images.githubusercontent.com/14109479/73468494-df3e4200-4352-11ea-8e18-5fbaa2fb945f.png">
<img width="766" alt="packaged" src="https://user-images.githubusercontent.com/14109479/73468502-e2393280-4352-11ea-9b31-7a09a872a8a2.png">

**Comment by Slapbox** [comment]
@uotw, for completeness, what OS and what version of `Electron-Builder`?

**Comment by uotw** [comment]
> @uotw, for completeness, what OS and what version of `Electron-Builder`?
> 
> I'm also seeing that I've lost subpixel antialiasing in a few places and only have grayscale antialiasing - which I'm (slowly) investigating now.
> 
> Neither of your images includes subpixel antialiasing, but only grayscale, which might be something to look into. If you're using the `css` `filter` property, that will affect antialiasing.


MacOS 10.15.2, "electron-packager": "^14.1.1", no filter property in css.  I have since found that on a different project, same MacOS/electron/electron-packager version I'm seeing no blurriness at all.  Only difference between the projects that I can see is that window.devicePixelRatio is getting reported as 2 for the sharp version and 1 for the blurry version.   Setting window.devicePixelRatio=2 in renderer.js doesn't change anything.  Also, the entire window is blurry, including dev tools:

<img width="338" alt="Screen Shot 2020-01-30 at 3 34 41 PM" src="https://user-images.githubusercontent.com/14109479/73487993-107c3980-4376-11ea-9fd5-05cd98cb9140.png">

**Comment by heronils** [comment]
I do not think this is related to subpixel antialiasing. The differences are too extreme. There are quite a few results on goofle regarding [MacOS blurry font](https://www.google.com/search?q=MacOS+blurry+font).

[Some](https://github.com/atom/atom/issues/17486#issuecomment-401600266) talk about using `-webkit-font-smoothing: subpixel-antialiased;` in the css.

[some](https://github.com/microsoft/vscode/issues/51132#issuecomment-424132330) talk about running in the terminal: `defaults write -g CGFontRenderingFontSmoothingDisabled -bool NO`.

Unfortunately i am not on a Mac (and know nothing about them), so i can not help further here.

**Comment by uotw** [comment]
> I do not think this is related to subpixel antialiasing. The differences are too extreme. There are quite a few results on goofle regarding [MacOS blurry font](https://www.google.com/search?q=MacOS+blurry+font).
> 
> [Some](https://github.com/atom/atom/issues/17486#issuecomment-401600266) talk about using `-webkit-font-smoothing: subpixel-antialiased;` in the css.
> 
> [some](https://github.com/microsoft/vscode/issues/51132#issuecomment-424132330) talk about running in the terminal: `defaults write -g CGFontRenderingFontSmoothingDisabled -bool NO`.
> 
> Unfortunately i am not on a Mac (and know nothing about them), so i can not help further here.

I tend to agree with you.  Something idiosyncratic about this particular app too, not all electron apps on my platform.  Disregard my issue.  Thanks

**Comment by uotw** [comment]
If anyone should happen upon this, I tracked down the issue.  MacOS had cached some electron settings in /Users/me/Library/Application Support/myapp/, so when I built my new app it was reading old settings. This problem does not apply when the app is located in /Applications, ostensibly because MacOS pays attention to versioning?  Not exactly sure.  Either way, you can fix the issue several ways: 

- delete the settings folder
- rename your project directory
- move the built app to a different directory (/Applications)

**Comment by Slapbox** [comment]
I've noticed that after upgrading to Electron@11.x from Electron 8.x, the subpixel antialiasing has become brutally hard to read for us. I'm trying to offer users a way to disable it if they desire, but there's no way to set a transparent background color for non-transparent windows. Is there anything we can do to force whole-pixel antialiasing to make our app legible again for our users?

**Comment by Slapbox** [comment]
Bump. 

A lot of things looks really crummy in our app. Surely there's some way to disable subpixel-antialiasing?

**We're not the only one seeing this issue:**
From https://github.com/BabylonJS/Editor/issues/266

> still found no definitive information on internet on how I can fix it on my side. This is definitely an issue for Electron.

Same experience here. I've searched and searched, tried all I can think of, but found no resolution for this. Can anybody suggest a way to optionally disable subpixel-antialiasing?

**Comment by Slapbox** [comment]
Bump. 

A lot of things looks really crummy in our app. Surely there's some way to disable subpixel-antialiasing?

I've searched and searched, tried all I can think of, but found no resolution for this. Can anybody suggest a way to optionally disable subpixel-antialiasing?

**Comment by nornagon** [comment]
@Slapbox would you be able to open a new issue with a minimal gist that shows the issue? Thanks!

**Comment by Slapbox** [comment]
It's weird, it seems like maybe the blurriness occurs when the system runs low on memory on Windows. I don't know why that would do it though. Not actionable for now.

**Comment by RJVB** [comment]
> I've noticed that after upgrading to [Electron@11.x](mailto:Electron@11.x) from Electron 8.x, the subpixel antialiasing has become brutally hard to read for us. 

Same thing with Franz; see https://github.com/meetfranz/franz/issues/1923 (annoyingly the fix no longer works with Franz v5.7.0 :( )

**Comment by Qiming-Liu** [comment]
![image](https://user-images.githubusercontent.com/68600416/145719966-965e1d5e-e727-48f8-ac00-7b23febfdccc.png)
